source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java

```diff
@@ -1,75 +1,204 @@
-
-    // Health text
-    int health = entity.getComponent(CombatStatsComponent.class).getHealth();
-    CharSequence healthText = String.format("Health: %d", health);
-    healthLabel = new Label(healthText, skin, "large");
-
-    table.add(heartImage).size(heartSideLength).pad(5);
-    table.add(healthLabel);
-    stage.addActor(table);
-  }
-
-  @Override
-  public void draw(SpriteBatch batch)  {
-    // draw is handled by the stage
-  }
-
-  /**
-   * Updates the player's health on the ui.
-   * @param health player health
-   */
-  public void updatePlayerHealthUI(int health) {
-    CharSequence text = String.format("Health: %d", health);
-    healthLabel.setText(text);
-  }
-
-  @Override
-  public void dispose() {
-    super.dispose();
-    heartImage.remove();
-    healthLabel.remove();
-  }
+    Table table;
+    private Image heartImage;
+    private Image xpImage;
+    private Image hungerImage;
+    private Label healthLabel;
+    private Label xpLabel;
+    private Label hungerLabel;
+    private Animation<TextureRegion> healthBarAnimation;
+    private Animation<TextureRegion> hungerBarAnimation;
+    private TextureAtlas[] textureAtlas;
+
+
+    /**
+     * Creates reusable ui styles and adds actors to the stage.
+     */
+    @Override
+    public void create() {
+        super.create();
+        addActors();
+        entity.getEvents().addListener("updateHealth", this::updatePlayerHealthUI);
+        entity.getEvents().addListener("updateExperience", this::updatePlayerExperienceUI);
+        entity.getEvents().addListener("updateHunger", this::updatePlayerHungerUI);
+    }
+
+    /**
+     * Initialises all required variables for health/xp/hunger bars
+     *
+     * @see Animation for animation details
+     */
+    public void initBarAnimations() {
+        // Initialise textureAtlas for 3 bars
+        textureAtlas = new TextureAtlas[3];
+        // HealthBar initialisation
+        textureAtlas[0] = new TextureAtlas("spriteSheets/healthBars.txt");
+        TextureRegion[] healthBarFrames = new TextureRegion[11];
+        // Names each frame and locates associated frame in txt file
+        for (int i = 0; i < healthBarFrames.length; i++) {
+            String frameName = (100 - i * 10) + "%_health";
+            healthBarFrames[i] = textureAtlas[0].findRegion(frameName);
+        }
+        healthBarAnimation = new Animation<>(0.066f, healthBarFrames);
+
+        // HungerBar initialisation
+        textureAtlas[1] = new TextureAtlas("images/hungerbar.atlas");
+        TextureRegion[] hungerBarFrames = new TextureRegion[11];
+        // Names each frame and locates associated frame in txt file
+        for (int i = 0; i < hungerBarFrames.length; i++) {
+        String frameName = (100 - i * 10) + "%_hunger";
+        hungerBarFrames[i] = textureAtlas[1].findRegion(frameName);
+        }
+        hungerBarAnimation = new Animation<>(0.066f, hungerBarFrames);
+
+        // xpBar initialisation
```

Comment:
if this is under development move it into a feature branch, should not be in main